aspnetcore/signalr/configuration.md

```diff
@@ -0,0 +1,202 @@
+
+The `HubConnectionBuilder` API is available for both C# and TypeScript clients.
+
+| Option | Description |
+| ------ | ----------- |
+| Headers | HTTP headers to be applied to all HTTP requests. |
+| Cookies | Cookies to be sent with each HTTP request. |
+| ClientCertificates | TLS client certificates to send when connecting over HTTP (not supported in Xamarin) |
+
+The following code samples demonstrate the C# setting connection options with the `HubConnectionBuilder`:
+
+```csharp
+// Creates a connection and restricts transports to WebSockets and Server Sent Events.
+var connection = new HubConnectionBuilder()
+    .WithUrl("url", HttpTransportType.WebSockets | HttpTransportType.ServerSentEvents)
+    .Build();
+
+// Creates a connection with a JSON Web Token.
+var connection = new HubConnectionBuilder()
+    .WithUrl("url", options => {
+        options.AccessTokenProvider = async () => {
+            // Get access token and return it.
+        };
+    });
+    .Build();
+
+// Creates a connection, and sets headers, cookies, and client certificates.
+var connection = new HubConnectionBuilder()
+    .WithUrl("url", options => {
+        options.Headers["Foo"] = "Bar";
+        options.Cookies.Add(new Cookie(...));
+        options.ClientCertificates.Add(...);
+    });
+    .Build();
+```
+
+The following code samples demonstrate setting connection options with the `HubConnectionBuilder` in JavaScript:
+
+```javascript
+// Creates a basic connection
+const connection = new signalR.HubConnectionBuilder()
+    .withUrl("url")
+    .build();
+
+// Sets the transport type. Transport types are defined in preceding table
+const connection = new signalR.HubConnectionBuilder()
+    .withUrl("url", HttpTransportType.WebSockets)
+    .build();
+
+// Sets the protocol such as Messagepack
+const connection = new signalR.HubConnectionBuilder()
+    .withUrl("url")
+    .withHubProtocol(IHubProtocol)
+    .build();
+```
+
+> ![NOTE]
+> Headers, cookies and client certificates cannot be configured in the JavaScript client due to limitations on browser APIs.
+
+The following table and code sample demonstrate the available `HttpConnectionOptions`.
+
+| Option | Description |
+| ------ | ----------- |
+| `httpClient`  | Class that handles all the GET, POST, and DELETE requests sent by the client. Can override with custom settings. |
+| `transport`  | Specific transport the client should use, or an `ITransport` implementation for a custom transport. |
+| `accessTokenFactory`  | Called for each HTTP request to set the authorization header or for WebSockets to set the `access_token` query string value. |
+| `logMessageContent`  | Log the message content when sending and receiving. Disabled by default. |
+| `skipNegotiation`  | Only use this when `HttpTransportType.WebSockets` is specified. It skips the negotiation step when it not necessary. |
```

Comment:
"It skips the negotiation step when it not necessary." =>
"It skips the negotiation step when it isn't necessary."